apps/worker-service/src/app/workflow/services/workflow-queue.service.spec.ts

```diff
@@ -174,21 +215,28 @@ describe('Workflow Queue service', () => {
 
     const jobCreated = await jobRepository.create(job);
 
-    await workflowQueueService.addToQueue(jobCreated._id, jobCreated, 0);
+    await queueService.addToQueue(jobCreated._id, jobCreated, '0');
 
-    await session.awaitRunningJobs(_templateId, false, 1);
+    await jobsService.awaitRunningJobs({
+      templateId: _templateId,
+      organizationId: organization._id,
+      delay: false,
+    });
     // We pause the worker as little trick to allow the `failed` status to be updated in the callback of the worker and not having a race condition.
-    await workflowQueueService.gracefulShutdown();
-
-    const jobs = await jobRepository.find({ _environmentId, _organizationId });
-    expect(jobs.length).to.eql(1);
-
-    expect(jobs[0].status).to.eql(JobStatusEnum.FAILED);
-    expect(jobs[0].error).to.deep.include({
-      status: 400,
+    await queueService.gracefulShutdown();
+
+    let failedTrigger: JobEntity | null = null;
+    do {
+      failedTrigger = await jobRepository.findOne({
+        _environmentId,
+        _organizationId,
+        status: JobStatusEnum.FAILED,
+        type: StepTypeEnum.TRIGGER,
+      });
+    } while (!failedTrigger || !failedTrigger.error);
```

Comment:
Should we maybe do some retry  count here? Just to make sure it's not running forever in the CI?

Reply:
the test will basically fail after the timeout of 10s